### ISSUE TITLE
Providing training loss to callback at end of epoch

### ISSUE THREAD

**Issue by pkch** [fchollet/keras#1118]
Is there any reason not to provide to the callback functions the training loss (and accuracy if accuracy monitoring is enabled) at the end of each epoch? This is only provided in the log at the end of each batch. 

This means (among other things) that EarlyStopping callback can't use 'loss' as the monitor; I understand why 'val_loss' is usually better, but there might be reasons to minimize the training loss.

Somewhat related to this, it may be worth documenting whether the 'loss' reported at the end of each batch is actually the loss calculated for the entire training dataset or just for that batch.

**Comment by jstypka** [comment]
I was wondering about exactly the same thing. It's difficult to access the `loss` and `acc` at the end of the epoch and those metrics are quiet likely to be monitored/logged
